piff/hyperpsf.py

```diff
@@ -0,0 +1,277 @@
+        self._model_comparer = model_comparer
+        self._model_comparer_weights = model_comparer_weights
+        self._model_init = model_init
+        self._interp_init = interp_init
+        self._model_error = model_error
+        self._interp_error = interp_error
+        self._model_limit = model_limit
+        self._interp_limit = interp_limit
+
+        # build the iminuit arguments
+        self._fit_func_kwargs_minuit = {}
+        Nmodel = len(self._model_keys)
+        Ninterp = len(self._interp_keys)
+        Nkeys = Nmodel + Ninterp
+        # in principle I could make these as long as I wanted.
+        if Nkeys > 50:
+            raise Exception('Max of 50 variables! You have {0}!'.format(Nkeys))
+        for i in range(Nkeys, 50):
+            # fix unused parameters
+            self._fit_func_kwargs_minuit['fix_p{0}'.format(i)] = True
+
+        # set initial values
+        if self._model_init:
+            for i, val in enumerate(self._model_init):
+                self._fit_func_kwargs_minuit['p{0}'.format(i)] = val
+        if self._interp_init:
+            for i, val in enumerate(self._interp_init):
+                self._fit_func_kwargs_minuit['p{0}'.format(i + Nmodel)] = val
+        # set rest
+        for i in xrange(Nkeys, 50):
+            self._fit_func_kwargs_minuit['p{0}'.format(i)] = 0
+
+        # set initial step sizes
+        if self._model_error:
+            for i, val in enumerate(self._model_error):
+                self._fit_func_kwargs_minuit['error_p{0}'.format(i)] = val
+        if self._interp_error:
+            for i, val in enumerate(self._interp_error):
+                self._fit_func_kwargs_minuit['error_p{0}'.format(i + Nmodel)] = val
+
+        # set limits
+        if self._model_limit:
+            for i, val in enumerate(self._model_limit):
+                self._fit_func_kwargs_minuit['limit_p{0}'.format(i)] = val
+        if self._interp_limit:
+            for i, val in enumerate(self._interp_limit):
+                self._fit_func_kwargs_minuit['limit_p{0}'.format(i + Nmodel)] = val
+
+        # run initial solves for model and interp
+        self.model.solve(self._stars)
+        self.interp.solve(self._stars)
+
+    # forgive me oh lord this python sin
+    def _fit_func(self, p0, p1, p2, p3, p4, p5, p6, p7, p8, p9,
+                  p10, p11, p12, p13, p14, p15, p16, p17, p18, p19,
+                  p20, p21, p22, p23, p24, p25, p26, p27, p28, p29,
+                  p30, p31, p32, p33, p34, p35, p36, p37, p38, p39,
+                  p40, p41, p42, p43, p44, p45, p46, p47, p48, p49,
+                  ):
+        # convert p to params
+        params = [p0, p1, p2, p3, p4, p5, p6, p7, p8, p9,
+                  p10, p11, p12, p13, p14, p15, p16, p17, p18, p19,
+                  p20, p21, p22, p23, p24, p25, p26, p27, p28, p29,
+                  p30, p31, p32, p33, p34, p35, p36, p37, p38, p39,
+                  p40, p41, p42, p43, p44, p45, p46, p47, p48, p49,]
```

Comment:
I looked into this a little bit, and I at least see where you are coming from here.  However, I think iminuit has ways around this problem.  The [Tutorial](http://nbviewer.jupyter.org/github/iminuit/iminuit/blob/master/tutorial/tutorial.ipynb#Faking-a-function-signature) describes ways to fake a function signature.  In particular, it looks like [`forced_parameters`](http://nbviewer.jupyter.org/github/iminuit/iminuit/blob/master/tutorial/tutorial.ipynb#Last-Resort:-Forcing-function-signature) might do what we want.

Please look into the solutions described here and see if you can get one of them to work.  The current solution is too ugly for me to want to leave it this way.